### PR TITLE
Don't wrap comments that are part of a table

### LINKED ISSUE #4210: `wrap_comments = true` breaks markdown tables
`cargo fmt -- --config wrap_comments=true` turns this:
```rust
/// doc
///
/// Table that is > 80 symbols:
///
/// | table | xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx |
/// |-------|--------------------------------------------------------------------|
/// | val   | x                                                                  |
struct Item;
```
image

Into this:
```rust
/// doc
///
/// Table that is > 80 symbols:
///
/// | table | xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
/// |
/// |-------|--------------------------------------------------------------------|
/// | val   | x
/// |
struct Item;
```
image

It's possible to work around this issue by adding `#[rustfmt::skip]` to the struct (or other documented item), but I think it's a bug anyway.

### PR DESCRIPTION
**Pull request by Nemo157**
Closes #4210

This is a trivial cherry-pick of https://github.com/rust-lang/rustfmt/pull/4214 onto the new `master` branch.

### CODE REVIEW

File: src/comment.rs
Comment (ytmimi) [inline src/comment.rs:943]: typo

```suggestion
/// Returns true if the given string may be part of a Markdown table.
```

File: tests/target/issue-4210.rs
Comment (ytmimi) [inline tests/target/issue-4210.rs:15]: I think these are good tests ported over from the backport, however I'd like to build on these before merging.

I'd like to see a test case for `wrap_comments=false` (just for completeness). Additionally, I'd like to add some text before the table that gets wrapped when `wrap_comments=true` to further emphasize that the table isn't being wrapped even when the text is wrapped.

I also think we should add a row to the table (besides the header) that contains a long line to show that it also isn't wrapped.

**Review by calebcartwright — APPROVED** [review]

**Comment by l4l** [comment]
Any chance of merging this one?

**Comment by ytmimi** [comment]
@l4l we plan to merge this and all other PRs labeled as `pr-ready-to-merge`. We're holding off on merging PRs right now because we haven't done a subtree sync with `rust-lang/rust` in a while and want to get the sync in before merging PRs. Hopefully we'll be able to get this merged soon!

**Review by calebcartwright — APPROVED** [review]

**Pull request merged**
